components/core/src/clp_s/indexer/IndexManager.hpp

```diff
@@ -38,20 +38,20 @@ class IndexManager {
 
     // Constructors
     IndexManager(
+            std::string const& dataset_name,
             std::optional<clp::GlobalMetadataDBConfig> const& db_config = std::nullopt,
             bool should_create_table = false
     );
 
     // Destructor
-    ~IndexManager();
+    ~IndexManager() = default;
 
     // Methods
     /**
      * Updates the metadata for a given archive
```

Comment:
```suggestion
     * Updates the metadata for a given archive.
```
Just while we're touching code around here anyway.